vic/drivers/shared_all/src/put_data.c

```diff
@@ -924,200 +848,209 @@ collect_eb_terms(energy_bal_struct energy,
+    out_data[OUT_SNOW_SURFT_BAND][band] += snow.surf_temp * Cv *
+                                           lakefactor;
 
     /** record pack layer temperature **/
-    out_data[OUT_SNOW_PACKT_BAND].data[band] += snow.pack_temp * Cv *
-                                                lakefactor;
+    out_data[OUT_SNOW_PACKT_BAND][band] += snow.pack_temp * Cv *
+                                           lakefactor;
 
     /** record latent heat of sublimation **/
-    out_data[OUT_LATENT_SUB_BAND].data[band] += energy.latent_sub * Cv *
-                                                lakefactor;
+    out_data[OUT_LATENT_SUB_BAND][band] += energy.latent_sub * Cv *
+                                           lakefactor;
 
     /** record band net downwards shortwave radiation **/
-    out_data[OUT_SWNET_BAND].data[band] += energy.NetShortAtmos * Cv *
-                                           lakefactor;
+    out_data[OUT_SWNET_BAND][band] += energy.NetShortAtmos * Cv *
+                                      lakefactor;
 
     /** record band net downwards longwave radiation **/
-    out_data[OUT_LWNET_BAND].data[band] += energy.NetLongAtmos * Cv *
-                                           lakefactor;
+    out_data[OUT_LWNET_BAND][band] += energy.NetLongAtmos * Cv *
+                                      lakefactor;
 
     /** record band albedo **/
     if (snow.snow && overstory) {
-        out_data[OUT_ALBEDO_BAND].data[band] += energy.AlbedoOver * Cv *
-                                                lakefactor;
+        out_data[OUT_ALBEDO_BAND][band] += energy.AlbedoOver * Cv *
+                                           lakefactor;
     }
     else {
-        out_data[OUT_ALBEDO_BAND].data[band] += energy.AlbedoUnder * Cv *
-                                                lakefactor;
+        out_data[OUT_ALBEDO_BAND][band] += energy.AlbedoUnder * Cv *
+                                           lakefactor;
     }
 
     /** record band net latent heat flux **/
-    out_data[OUT_LATENT_BAND].data[band] -= energy.latent * Cv * lakefactor;
+    out_data[OUT_LATENT_BAND][band] -= energy.latent * Cv * lakefactor;
 
     /** record band net sensible heat flux **/
-    out_data[OUT_SENSIBLE_BAND].data[band] -= energy.sensible * Cv * lakefactor;
+    out_data[OUT_SENSIBLE_BAND][band] -= energy.sensible * Cv * lakefactor;
 
     /** record band net ground heat flux **/
-    out_data[OUT_GRND_FLUX_BAND].data[band] -= energy.grnd_flux * Cv *
-                                               lakefactor;
+    out_data[OUT_GRND_FLUX_BAND][band] -= energy.grnd_flux * Cv *
+                                          lakefactor;
+}
+
+/******************************************************************************
+ * @brief    Initialize the structures used in put_data
+ *****************************************************************************/
+void
+initialize_put_data()
+{
+    // TODO: do required initialization here.
```

Comment:
This is still a todo.  We need to figure out what to do with accumulated storage and error terms.